composer.lock

```diff
@@ -244,36 +244,918 @@
             },
             "time": "2021-03-21T15:43:46+00:00"
         },
+        {
+            "name": "psr/container",
+            "version": "1.1.1",
+            "source": {
+                "type": "git",
+                "url": "https://github.com/php-fig/container.git",
+                "reference": "8622567409010282b7aeebe4bb841fe98b58dcaf"
+            },
+            "dist": {
+                "type": "zip",
+                "url": "https://api.github.com/repos/php-fig/container/zipball/8622567409010282b7aeebe4bb841fe98b58dcaf",
+                "reference": "8622567409010282b7aeebe4bb841fe98b58dcaf",
+                "shasum": ""
+            },
+            "require": {
+                "php": ">=7.2.0"
+            },
+            "type": "library",
+            "autoload": {
+                "psr-4": {
+                    "Psr\\Container\\": "src/"
+                }
+            },
+            "notification-url": "https://packagist.org/downloads/",
+            "license": [
+                "MIT"
+            ],
+            "authors": [
+                {
+                    "name": "PHP-FIG",
+                    "homepage": "https://www.php-fig.org/"
+                }
+            ],
+            "description": "Common Container Interface (PHP FIG PSR-11)",
+            "homepage": "https://github.com/php-fig/container",
+            "keywords": [
+                "PSR-11",
+                "container",
+                "container-interface",
+                "container-interop",
+                "psr"
+            ],
+            "support": {
+                "issues": "https://github.com/php-fig/container/issues",
+                "source": "https://github.com/php-fig/container/tree/1.1.1"
+            },
+            "time": "2021-03-05T17:36:06+00:00"
+        },
+        {
+            "name": "symfony/console",
+            "version": "v5.4.9",
```

Comment:
@jonasraoni, OJS 3.3.x already includes `symfony/console` (currently v5.2.2). I'm concerned that we're going to end up with difficult-to-diagnose problems if we have plugins including autoloaded competing versions of the same libraries. Could you take a quick look at Composer to see if it can e.g. lock dependencies based on an external `composer.lock`, or something equivalent?

Reply:
@asmecher The `symfony` package is used only by a command-line tool to validate bags, a single file separated from the rest of the code base. So, it's ok merge the PR.

---

The dependency hell is a tough problem to solve, and such plugins bring an extra dose of unpredictability 👀
I'll take a look at the options, at least to decrease the problem, but I'm already not very optimistic about it :)

Reply:
No silver bullet, but here are some related discussions.

https://stackoverflow.com/questions/44999348/multiple-composer-installs-with-independent-modular-projects
https://stackoverflow.com/questions/23647387/order-of-execution-of-multiple-composer-autoloaders

This issue will also crop up in the Wordpress ecosystem, where a set of practices has probably emerged:

https://wordpress.stackexchange.com/questions/175392/how-to-prevent-composer-dependency-conflicts-amongst-wp-plugins